controller/src/main/java/io/pravega/controller/fault/ControllerClusterListener.java

```diff
@@ -82,6 +87,9 @@ protected void startUp() throws Exception {
                         if (eventProcessorsOpt.isPresent() && eventProcessorsOpt.get().isRunning()) {
                             eventProcessorsOpt.get().notifyProcessFailure(host.getHostId());
                         }
+                        if (txnSweeperOpt.isPresent() && txnSweeperOpt.get().isReady()) {
```

Comment:
That would cause us to lose events arriving after the moved block but before the call to `Cluster.AddListener` gets executed.